pkg/controller/cluster/main-controller.go

```diff
@@ -342,6 +342,13 @@ func (c *Controller) syncHandler(key string) error {
 	mi.EnsureDefaults()
 	miniov1.InitGlobals(mi)
 
+	// Validate the MinIO Instance
+	if err = mi.Validate(); err != nil {
+		mi, err = c.updateMinIOInstanceStatus(ctx, mi, err.Error(), 0)
+		klog.V(2).Infof(err.Error())
```

Comment:
err can be `nil` @dvaldivia this may crash, is it guaranteed that? this error is not nil?

Reply:
the `if` is making sure `err != nil`